### PR TITLE
add basic metrics

### PR DESCRIPTION
**Pull request by dysinger**

### CODE REVIEW

**Comment by dysinger** [comment]
Results in metrics served up ready for prometheus:
```
user@computer:~/src/github.com/fpco/stackage-server% curl localhost:3000/metrics | grep -i stackage
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  3799    0  3799    0     0  1251k      0 --:--:-- --:--:-- --:--:-- 1854k
# HELP stackage_server_fn Stackage Server function call (duration in microseconds).
# TYPE stackage_server_fn summary
stackage_server_fn{fn="Handler.Feed.getBranchFeedR",quantile="0.5"} 663682.0
stackage_server_fn{fn="Handler.Feed.getBranchFeedR",quantile="0.9"} 663682.0
stackage_server_fn{fn="Handler.Feed.getBranchFeedR",quantile="0.99"} 663682.0
stackage_server_fn_sum{fn="Handler.Feed.getBranchFeedR"} 663682.0
stackage_server_fn_count{fn="Handler.Feed.getBranchFeedR"} 1
stackage_server_fn{fn="Handler.Hoogle.getHoogleDB",quantile="0.5"} 25.0
stackage_server_fn{fn="Handler.Hoogle.getHoogleDB",quantile="0.9"} 33.0
stackage_server_fn{fn="Handler.Hoogle.getHoogleDB",quantile="0.99"} 33.0
stackage_server_fn_sum{fn="Handler.Hoogle.getHoogleDB"} 126.0
stackage_server_fn_count{fn="Handler.Hoogle.getHoogleDB"} 5
stackage_server_fn{fn="Handler.Hoogle.getHoogleR",quantile="0.5"} 630789.0
stackage_server_fn{fn="Handler.Hoogle.getHoogleR",quantile="0.9"} 5243028.0
stackage_server_fn{fn="Handler.Hoogle.getHoogleR",quantile="0.99"} 5243028.0
stackage_server_fn_sum{fn="Handler.Hoogle.getHoogleR"} 1.120087e7
stackage_server_fn_count{fn="Handler.Hoogle.getHoogleR"} 4
stackage_server_fn{fn="Handler.Hoogle.runHoogleQuery",quantile="0.5"} 101.0
stackage_server_fn{fn="Handler.Hoogle.runHoogleQuery",quantile="0.9"} 132.0
stackage_server_fn{fn="Handler.Hoogle.runHoogleQuery",quantile="0.99"} 132.0
stackage_server_fn_sum{fn="Handler.Hoogle.runHoogleQuery"} 431.0
stackage_server_fn_count{fn="Handler.Hoogle.runHoogleQuery"} 4
stackage_server_fn{fn="Handler.Package.packagePage",quantile="0.5"} 16204.0
stackage_server_fn{fn="Handler.Package.packagePage",quantile="0.9"} 19443.0
stackage_server_fn{fn="Handler.Package.packagePage",quantile="0.99"} 19443.0
stackage_server_fn_sum{fn="Handler.Package.packagePage"} 35647.0
stackage_server_fn_count{fn="Handler.Package.packagePage"} 2
stackage_server_fn{fn="Handler.Snapshots.getAllSnapshotsR",quantile="0.5"} 5228.0
stackage_server_fn{fn="Handler.Snapshots.getAllSnapshotsR",quantile="0.9"} 28370.0
stackage_server_fn{fn="Handler.Snapshots.getAllSnapshotsR",quantile="0.99"} 28370.0
stackage_server_fn_sum{fn="Handler.Snapshots.getAllSnapshotsR"} 33598.0
stackage_server_fn_count{fn="Handler.Snapshots.getAllSnapshotsR"} 2
stackage_server_fn{fn="Handler.StackageHome.getStackageDiffR",quantile="0.5"} 142650.0
stackage_server_fn{fn="Handler.StackageHome.getStackageDiffR",quantile="0.9"} 142650.0
stackage_server_fn{fn="Handler.StackageHome.getStackageDiffR",quantile="0.99"} 142650.0
stackage_server_fn_sum{fn="Handler.StackageHome.getStackageDiffR"} 142650.0
stackage_server_fn_count{fn="Handler.StackageHome.getStackageDiffR"} 1
stackage_server_fn{fn="Handler.StackageHome.getStackageHomeR",quantile="0.5"} 67948.0
stackage_server_fn{fn="Handler.StackageHome.getStackageHomeR",quantile="0.9"} 73553.0
stackage_server_fn{fn="Handler.StackageHome.getStackageHomeR",quantile="0.99"} 73553.0
stackage_server_fn_sum{fn="Handler.StackageHome.getStackageHomeR"} 191684.0
stackage_server_fn_count{fn="Handler.StackageHome.getStackageHomeR"} 3
```

**Comment by chrisdone** [comment]
Tried it out on my end. Pretty cool!

```
Chriss-MBP:Work chris$ curl localhost:3000/metrics | grep -i stackage
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  4472    0  4472    0     0   656k      0 --:--:-- --:--:-- --:--:--  727k
# HELP stackage_server_fn Stackage Server function call (duration in microseconds).
# TYPE stackage_server_fn summary
stackage_server_fn{fn="Handler.Package.packagePage",quantile="0.5"} 306548.0
stackage_server_fn{fn="Handler.Package.packagePage",quantile="0.9"} 306548.0
stackage_server_fn{fn="Handler.Package.packagePage",quantile="0.99"} 306548.0
stackage_server_fn_sum{fn="Handler.Package.packagePage"} 306548.0
stackage_server_fn_count{fn="Handler.Package.packagePage"} 1
# HELP stackage_server_fn Stackage Server function call (duration in microseconds).
# TYPE stackage_server_fn summary
stackage_server_fn{fn="Handler.StackageHome.getStackageHomeR",quantile="0.5"} 372955.0
stackage_server_fn{fn="Handler.StackageHome.getStackageHomeR",quantile="0.9"} 372955.0
stackage_server_fn{fn="Handler.StackageHome.getStackageHomeR",quantile="0.99"} 372955.0
stackage_server_fn_sum{fn="Handler.StackageHome.getStackageHomeR"} 372955.0
stackage_server_fn_count{fn="Handler.StackageHome.getStackageHomeR"} 1
```

File: Handler/BuildPlan.hs [hunk 1]
```diff
@@ -7,7 +7,7 @@ import Stackage.BuildPlan
 import Stackage.Database
 
 getBuildPlanR :: SnapName -> Handler TypedContent
-getBuildPlanR slug = do
+getBuildPlanR slug = track "Handler.BuildPlan.getBuildPlanR" $ do
```
Comment (chrisdone) [inline Handler/BuildPlan.hs]: Perhaps in a future change, this could be automated with template-haskell.

File: Import.hs [hunk 2]
```diff
@@ -35,3 +37,26 @@ haddockUrl sname pkgver name = HaddockR sname
   where
     toDash '.' = '-'
     toDash c = c
+
+track
+    :: MonadIO m
+    => String -> m a -> m a
+track name inner = do
```
Comment (chrisdone) [inline Import.hs]: I think in future a more reliable way to get time would be via `Criterion.Measurement`:

http://hackage.haskell.org/package/criterion-1.1.1.0/docs/Criterion-Measurement.html

E.g.
``` haskell
do initializeTime
   start <- getCPUTime
   r <- inner
   end <- getCPUTime
```

File: Handler/BuildPlan.hs (hunk 1 above)
Comment (dysinger) [inline Handler/BuildPlan.hs]: agree. I wanted to go checkout $logXXXX examples next.

File: Import.hs (hunk 2 above)
Comment (dysinger) [inline Import.hs]: noted

**Pull request merged**